src/compiler/output-targets/output-utils.ts

```diff
@@ -99,40 +69,34 @@ export const isOutputTargetDistTypes = (o: d.OutputTarget): o is d.OutputTargetD
 export const getComponentsFromModules = (moduleFiles: d.Module[]) =>
   sortBy(flatOne(moduleFiles.map((m) => m.cmps)), (c: d.ComponentCompilerMeta) => c.tagName);
 
-export const canSkipOutputTargets = (buildCtx: d.BuildCtx) => {
-  if (buildCtx.components.length === 0) {
-    return true;
-  }
-  if (buildCtx.requiresFullBuild) {
-    return false;
-  }
-  if (buildCtx.isRebuild && (buildCtx.hasScriptChanges || buildCtx.hasStyleChanges || buildCtx.hasHtmlChanges)) {
-    return false;
-  }
-  return true;
-};
-
-export const ANGULAR = `angular`;
+export const ANGULAR = 'angular';
 export const COPY = 'copy';
-export const CUSTOM = `custom`;
-export const DIST = `dist`;
-export const DIST_COLLECTION = `dist-collection`;
-export const DIST_CUSTOM_ELEMENTS = `dist-custom-elements`;
-export const DIST_CUSTOM_ELEMENTS_BUNDLE = `dist-custom-elements-bundle`;
-
-export const DIST_TYPES = `dist-types`;
-export const DIST_HYDRATE_SCRIPT = `dist-hydrate-script`;
-export const DIST_LAZY = `dist-lazy`;
-export const DIST_LAZY_LOADER = `dist-lazy-loader`;
+export const CUSTOM = 'custom';
+export const DIST = 'dist';
+export const DIST_COLLECTION = 'dist-collection';
+export const DIST_CUSTOM_ELEMENTS = 'dist-custom-elements';
+export const DIST_CUSTOM_ELEMENTS_BUNDLE = 'dist-custom-elements-bundle';
+
+export const DIST_TYPES = 'dist-types';
+export const DIST_HYDRATE_SCRIPT = 'dist-hydrate-script';
+export const DIST_LAZY = 'dist-lazy';
+export const DIST_LAZY_LOADER = 'dist-lazy-loader';
 export const DIST_GLOBAL_STYLES = 'dist-global-styles';
 export const DOCS_CUSTOM = 'docs-custom';
-export const DOCS_JSON = `docs-json`;
-export const DOCS_README = `docs-readme`;
-export const DOCS_VSCODE = `docs-vscode`;
-export const STATS = `stats`;
-export const WWW = `www`;
-
-export const VALID_TYPES = [
+export const DOCS_JSON = 'docs-json';
+export const DOCS_README = 'docs-readme';
+export const DOCS_VSCODE = 'docs-vscode';
+export const STATS = 'stats';
+export const WWW = 'www';
+
+/**
+ * Valid output targets to specify in a Stencil config.
+ *
+ * Note that there are some output targets (e.g. `DIST_TYPES`) which are programmatically
+ * set as output targets by the compiler when other output targets (in that case `DIST`) are set, but which are _not_ supported in a Stencil config. This is enforced in the output
+ * target validation code.
+ */
+export const VALID_CONFIG_OUTPUT_TARGETS = [
```

Comment:
I renamed this because I didn't think `VALID_TYPES` was a very good name, especially for an identifier that is exported outside of the context of a single module (I think even for an unexported name I'd want something more descriptive but 🤷‍♀️ )